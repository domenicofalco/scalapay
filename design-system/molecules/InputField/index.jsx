import PropTypes from "prop-types";
import styled from 'styled-components'
import { Input as I } from "usetheform";
import { grey900 } from "colors";
import { small, bold } from "fonts";
import { tablet } from "breakpoints";

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;

  @media (min-width: ${tablet}) {
    flex-direction: row;
    align-items: center;
  }
`;

const Label = styled.label`
  font-size: ${small};
  font-weight: ${bold};
  margin-right: 10px;
  margin-bottom: 4px;

  @media (min-width: ${tablet}) {
    width: 30%;
    margin-bottom: 0;  
  }
`;

const Input = styled(I)`
  padding: 5px 12px;
  border-radius: 5px;
  border: 1px solid ${grey900};
  font-size: ${small};
  width: 100%;
`;

const InputField = ({type, label, ...rest}) => {
  return (
    <Wrapper>
      <Label>{label}</Label>
      <Input type={type} {...rest} />
    </Wrapper>
  );
}

InputField.defaultProps = {
  label: ""
};

InputField.propTypes = {
  type: PropTypes.string.isRequired,
  label: PropTypes.string,
};

export default InputField;