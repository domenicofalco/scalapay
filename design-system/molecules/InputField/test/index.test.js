import React from "react";
import * as hooks from "usetheform";
import { cleanup, render } from "@testing-library/react";
import InputField from "../";

afterEach(cleanup);

describe("molecules => InputField", () => {
  it("It should render the inputfield with a label", () => {
    const { container, getByText, asFragment } = render(
      <hooks.Form>
        <InputField label="a_label" type="text" />
      </hooks.Form>
    );
    const label = getByText(/a_label/i);
    expect(label).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  it("It should render the inputfield with no error", () => {
    jest
      .spyOn(hooks, "useValidation")
      .mockImplementation(() => [{ isValid: true }]);

    const { container } = render(
      <hooks.Form>
        <InputField label="a_label" type="text" />
      </hooks.Form>
    );
    const input = container.querySelector("input");
    const borderStyle = input.style.border;
    expect(borderStyle).toBe("2px solid #666666");
  });

  it("It should render the inputfield with error", () => {
    jest
      .spyOn(hooks, "useValidation")
      .mockImplementation(() => [{ isValid: false }]);

    const { container } = render(
      <hooks.Form>
        <InputField label="a_label" type="text" />
      </hooks.Form>
    );
    const input = container.querySelector("input");
    const borderStyle = input.style.border;
    expect(borderStyle).toBe("2px solid #ff0000");
  });
});
