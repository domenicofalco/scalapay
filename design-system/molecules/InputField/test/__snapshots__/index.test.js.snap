// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules => InputField It should render the inputfield with a label 1`] = `
<DocumentFragment>
  <form>
    <div
      class="InputField__Wrapper-sc-53rc5u-0 gMZqUu"
    >
      <label
        class="InputField__Label-sc-53rc5u-1 kwFgpj"
      >
        a_label
      </label>
      <input
        class="InputField__Input-sc-53rc5u-2 csGXuB"
        style="border: 2px solid #666666;"
        type="text"
        value=""
      />
    </div>
  </form>
</DocumentFragment>
`;
